src/main/java/io/reactivex/Observable.java

```diff
@@ -317,15 +317,17 @@ public static int bufferSize() {
         return lift(new OperatorTake<>(n));
     }
     
-    public <U> Observable<T> takeUntil(Publisher<U> other) {
+    public final <U> Observable<T> takeUntil(Publisher<U> other) {
+        Objects.requireNonNull(other);
         return lift(new OperatorTakeUntil<>(other));
     }
     
-    public Observable<T> takeUntil(Predicate<? super T> predicate) {
+    public final Observable<T> takeUntil(Predicate<? super T> predicate) {
+        Objects.requireNonNull(predicate);
         return lift(new OperatorTakeUntilPredicate<>(predicate));
     }
     
-    public Observable<T> takeLast(int n) {
+    public final Observable<T> takeLast(int n) {
```

Comment:
See #1353 for a suggestion for consistent naming of parameters that indicate a count of items to be emitted across operators with such parameters.